catalog/spatial/kml/spatial-kml-transformer/src/main/java/org/codice/ddf/spatial/kml/transformer/KmzTransformer.java

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.spatial.kml.transformer;
+
+import ddf.catalog.data.BinaryContent;
+import ddf.catalog.transform.MetacardTransformer;
+import ddf.catalog.transform.QueryResponseTransformer;
+
+/**
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library. </b> The {@link KmzTransformer} is extends the {@link
+ * QueryResponseTransformer} and {@link MetacardTransformer} to perform KMZ transformations on query
+ * results and single metacards.
+ */
+public interface KmzTransformer extends QueryResponseTransformer, MetacardTransformer {
```

Comment:
❓ If we're not registering this as a service (and not using the interface anywhere) could we just have the class extend these two interfaces itself so we can remove this file?

Reply:
I created this as an interface so it could be later registered as a service and the interface could be extended to support more advanced KMZ features, like stripping resources from a QueryResponse or Metacard before transforming to KML, adding the relative paths, and including them in a supporting documents folder in the resulting KMZ file.

Reply:
Can we still accomplish the more advanced features within the existing KmzTransformerImpl class? If we're changing the interface when we need to update functionality, then it doesn't feel like a contract as much as a copy of the class methods. (Also extending QueryResponseTransformer and MetacardTransformer seems like we are making this interface cover a lot of things.) If we need generic TypeA to TypeB transformers this means we would require creating a new interface for every single type conversion we need done. If there's a need later (to do this type of thing for many others) we could always create a generic conversion service at that time where there's a single common interface and provide the source and target types to the single service to do the transformation. (Eg, HTTP has a pattern similar to this- Accept/Accept-Encoding and Content-Type/Content-Encoding) 

If there is not a need now and we're unsure if further work will be needed outside of just changes to the implementation class we can always add it when its needed as opposed to putting it here now and risking it being unused.

Reply:
❗️ This is still unused